tests/src/FunctionalJavascript/CorporateContactFormTest.php

```diff
@@ -0,0 +1,320 @@
+    'corporate_fields[optional_fields][oe_telephone]' => TRUE,
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'user',
+    'oe_contact_forms',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Create and login test user with permission to create contact forms.
+    /** @var \Drupal\user\UserInterface $testuser */
+    $testuser = $this->drupalCreateUser([
+      'administer contact forms',
+    ]);
+    $this->drupalLogin($testuser);
+  }
+
+  /**
+   * Tests the corporate contact form.
+   */
+  public function testCorporateContactForm(): void {
+    /** @var \Behat\Mink\WebAssert $assert */
+    $assert = $this->assertSession();
+    /** @var \Behat\Mink\Element\DocumentElement $page */
+    $page = $this->getSession()->getPage();
+
+    $this->drupalGet('admin/structure/contact/add');
+
+    // Assert corporate fields are not present before is_corporate_form click.
+    $is_corporate_form = $page->findField('is_corporate_form');
+    $this->assertNotEmpty($is_corporate_form);
+    $this->assertFieldsAreMissing();
+
+    // Ajax call to load corporate fields.
+    $is_corporate_form->click();
+    $assert->assertWaitOnAjaxRequest();
+
+    // Assert fields are now visible.
+    $this->assertFieldsAreVisible();
+
+    // Assert expose_as_block is checked by default.
+    $element = $page->findField('corporate_fields[expose_as_block]');
+    $this->assertNotEmpty($element);
+    $this->assertTrue($element->isChecked());
+
+    // Add contact required values.
+    $this->fillCoreContactFields();
+
+    // Test topic ajax.
+    $this->assertTopicAddAjax();
+
+    // Test remove topic group.
+    $this->assertTopicRemoveAjax();
+
+    // Add remaining field values and submit form.
+    $this->fillCorporateFields();
+    $page->pressButton('Save');
+    $assert->pageTextContains('Contact form Test form has been added.');
+
+    // Assert the values are saved.
+    $this->drupalGet('admin/structure/contact/manage/oe_corporate_form');
+    $is_corporate_form = $page->findField('is_corporate_form');
+    $this->assertNotEmpty($is_corporate_form);
+    $this->assertTrue($is_corporate_form->isChecked());
+
+    // Make sure the saved values are the ones expected.
+    $this->checkCorporateFieldsOnPage();
+    $this->checkCorporateFieldsInStorage();
+  }
+
+  /**
+   * Tests the corporate contact form.
+   */
+  public function testNoCorporateValues(): void {
+    /** @var \Behat\Mink\WebAssert $assert */
+    $assert = $this->assertSession();
+    /** @var \Behat\Mink\Element\DocumentElement $page */
+    $page = $this->getSession()->getPage();
+
+    $this->drupalGet('admin/structure/contact/add');
+
+    // Assert corporate fields are not present.
+    $is_corporate_form = $page->findField('is_corporate_form');
+    $this->assertNotEmpty($is_corporate_form);
+    $this->assertFieldsAreMissing($page);
```

Comment:
There is no parameter on `assertFieldsAreMissing` Check all uses.

Reply:
fixed